src/H5Gent.c

```diff
@@ -160,6 +175,9 @@ H5G_ent_decode(const H5F_t *f, const uint8_t **pp, H5G_entry_t *ent)
 
     *pp = p_ret + H5G_SIZEOF_ENTRY_FILE(f);
 
+    if (H5_IS_BUFFER_OVERFLOW(*pp, 1, p_end))
+        HGOTO_ERROR(H5E_FILE, H5E_OVERFLOW, FAIL, "image pointer is out of bounds")
```

Comment:
Looks like this check is triggering in a lot of places.

Reply:
well, in fact, this check is after the last dereference, maybe it is not needed here

should I remove it?